sdk/cosmosdb/cosmos/src/CosmosClientOptions.ts

```diff
@@ -28,6 +28,8 @@ export interface CosmosClientOptions {
    * Allows users to generating their own auth tokens, potentially using a separate service
    */
   tokenProvider?: TokenProvider;
+  /** AAD token from @azure/identity */
+  aadToken?: string;
```

Comment:
Can we accept the full credential object from @azure/identity instead? I think that way it will handle refreshes for us.

Reply:
@southpolesteve updated - let me know if you like this approach more. I still need to add a TS type for aadCredentials